e2e/queryit.py

```diff
@@ -147,6 +147,18 @@ def test_check(check_dict: Dict[str, Any], result: Dict[str, Any]) -> bool:
                 eprint("contains_row check failed:\n" +
                        "\tdid not find %r" % gold_row)
                 return False
+        elif check == 'contains_warning':
+            for requested_warning in value:
+                found = False
+                for actual_warning in result["warnings"]:
+                    if actual_warning.startswith(requested_warning):
```

Comment:
`.startswith()` is already a bit fuzzy, why not just do `.contains()` and then it is all about containing a warning